### PR TITLE
Workbench

### LINKED ISSUE #2: Light bulb
- [x] Create light bulb model
 - [x] Create shader for filament glow (dimming)
 - [x] Add to scene to workbench as a selectable object

### LINKED ISSUE #5: Workbench
- [x] Create workbench model
 - [x] Add workbench to scene
 - [x] Enable storing items in the workbench

### PR DESCRIPTION
**Pull request by raaavioli**
Adds light and workbench to scene. Completely solves #2 as well as #5.

Items can be added to workbench by drag-dropping the item prefabs into the "Tool Object Prefabs"-list in the ToolBoard.

## Dimmable lamps
lamp

## Workbench with storage
image

closing #5
closing #2

### CODE REVIEW

File: Connected/Assets/Materials/PlasticBlack.mat [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    - _Mode: 0
+    - _OcclusionStrength: 1
+    - _Parallax: 0.02
+    - _SmoothnessTextureChannel: 0
+    - _SpecularHighlights: 1
+    - _SrcBlend: 1
+    - _UVSec: 0
+    - _ZWrite: 1
+    m_Colors:
+    - _Color: {r: 1, g: 1, b: 1, a: 1}
```
Comment (steenish) [inline Connected/Assets/Materials/PlasticBlack.mat]: Should this be black rather than white?

File: Connected/Assets/Scripts/Dimming.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private void Start()
+    {
+        FilamentMpb = new MaterialPropertyBlock();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        FilamentMpb.SetColor("_EmissionColor", filamentColor * Intensity);
+        Filament.GetComponent<Renderer>().SetPropertyBlock(FilamentMpb);
```
Comment (steenish) [inline Connected/Assets/Scripts/Dimming.cs]: GetComponent is too slow to be called in Update, you can store the renderer component in a private field, which you set in Awake.

File: Connected/Assets/Scripts/Dimming.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    Light LightSource;
+
+    [SerializeField]
+    [Range(0.0f, 20.0f)]
+    float Intensity = 1.0f;
+    float MaxIntensity = 20.0f;
+
+    private Color filamentColor = new Color(0.7490196f, 0.427451f, 0);
+
+    private void Start()
```
Comment (steenish) [inline Connected/Assets/Scripts/Dimming.cs]: This should probably be Awake, not Start

File: Connected/Assets/Scripts/ToolBoard.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        InactiveMPB.SetColor("_Color", Colors.DarkBrown);
+        ActiveMPB = new MaterialPropertyBlock();
+        ActiveMPB.SetColor("_Color", Colors.Orange);
+    }
+
+    public void Update()
+    {
+
+        if (toolPrefab != null)
+            marker.GetComponent<Renderer>().SetPropertyBlock(ActiveMPB);
```
Comment (steenish) [inline Connected/Assets/Scripts/ToolBoard.cs]: GetComponent should not be done in Update, the renderer component can be stored in a private field that is set in the constructor

File: Connected/Assets/Scripts/ToolBoard.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        else
+            marker.GetComponent<Renderer>().SetPropertyBlock(InactiveMPB);
```
Comment (steenish) [inline Connected/Assets/Scripts/ToolBoard.cs]: ^

**Review by steenish — CHANGES_REQUESTED** [review]
most of it looks good, just some notes

File: Connected/Assets/Materials/PlasticBlack.mat (hunk 1 above)
Comment (raaavioli) [reply Connected/Assets/Materials/PlasticBlack.mat]: I don't know why this is. In the editor it's set to {0.0943396, 0.0943396, 0.0943396. 1}, as far as I can see and the color of the splitter using that color is not appearing white. Am I missing something? 

The changes were applied in my editor, but did not reflect in git until i tweaked some other parameters aswell. The editor did not say that I had unsaved changes, and restarting the application would not do anything either. Odd.

**Comment by raaavioli** [comment]
This should be it :)

**Review by steenish — APPROVED** [review]
lgtm

**Pull request merged**